authors/apps/comments/views.py

```diff
@@ -0,0 +1,129 @@
+            comment['article'] = article.first().id
+            serializer = CommentSerializer(data=comment)
+            serializer.is_valid(raise_exception=True)
+            serializer.save(
+                author=self.request.user
+            )
+            return Response({
+                "comment": serializer.data,
+                "message": "Comment successfully created."
+            }, status=status.HTTP_201_CREATED)
+        return Response({
+            "error": "Article not found."
+        }, status=status.HTTP_404_NOT_FOUND)
+
+    def get(self, request, **kwargs):
+        """
+        This method returns a list of comments on an article.
+        """
+        article = Article.objects.filter(id=kwargs['article_id'])
+        if article:
+            queryset = Comment.objects.filter(
+                article=self.kwargs['article_id'])
+            serializer = CommentSerializer(queryset, many=True)
+            return Response({
+                "comments": serializer.data
+            }, status=status.HTTP_200_OK)
+        return Response({
+            "error": "Article not found."
+        }, status=status.HTTP_404_NOT_FOUND)
+
+
+class RetrieveUpdateDeleteComment(APIView):
+    """
+    GET Comment/
+    Update Comment/
+    Delete Comment/
+    """
+    queryset = Comment.objects.all()
+    serializer_class = CommentSerializer
+
+    @classmethod
+    def get_object(cls, pk):
+        try:
+            return Comment.objects.get(pk=pk)
+        except Comment.DoesNotExist:
+            raise NotFound({"error": "Comment not found."})
+
+    def get(self, request, pk, **kwargs):
+        """
+        This method returns a single comment.
+        """
+        article = Article.objects.filter(id=kwargs['article_id'])
+        if article:
+            comment = self.get_object(pk)
+            serializer = CommentSerializer(comment)
+            return Response({"comment": serializer.data})
+        return Response({
+            "error": "Article not found."
+        }, status=status.HTTP_404_NOT_FOUND)
+
+    def put(self, request, pk, **kwargs):
+        """
+        This method updates a single comment.
+        """
+        article = Article.objects.filter(id=kwargs['article_id'])
+        if article:
+            comment = self.get_object(pk)
+            if comment.author.username == request.user.username:
+                data = {}
+                data['article'] = article.first().id
+                data['body'] = request.data['comment']['body']
+                serializer = CommentSerializer(comment, data=data)
+                serializer.is_valid(raise_exception=True)
+                serializer.save()
+                return Response({
+                    "comment": serializer.data,
+                    "message": "Comment successfully updated."
+                }, status=status.HTTP_200_OK)
+            return Response({
+                "error": "You cannot edit this comment."
```

Comment:
I believe you should add an easier to understand response message here
```suggestion
                "error": "You do not have permission to edit this comment because you did not create it."
```